### PR TITLE
CIV-13172 Lip Email Notification Determination of Means

### PR DESCRIPTION
**Pull request by Omaira-Melo-Hmcts**
### JIRA link (if applicable) ###
[Email notifications: Claimant/Defendant LiP - Judgment varied following Determination of Means](https://tools.hmcts.net/jira/browse/CIV-13172)

### Change description ###
 Generate an email notification for Claimant and Defendant LiP after a Judgment has been varied following a Determination of Means
image

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyClaimantJudgmentVariedDeterminationOfMeansNotificationHandlerTest.java [hunk 1]
```diff
@@ -85,6 +93,32 @@ void shouldNotifyClaimantJudgmentVariedDeterminationOfMeans_whenInvoked() {
...
             );
         }
+
+        @Test
+        void shouldNotifyApplicantLip_whenInvoked() {
+            CaseData caseData = CaseDataBuilder.builder().buildJudgmentOnlineCaseDataWithDeterminationMeans();
+            caseData = caseData.toBuilder()
+                .applicant1(Party.builder()
+                                .individualFirstName("Applicant1").individualLastName("ApplicantLastName").partyName("Applicant1")
+                                .type(Party.Type.INDIVIDUAL).partyEmail("respondentLip@example.com").build())
```
Comment (hmcts-version1-ignacio) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyClaimantJudgmentVariedDeterminationOfMeansNotificationHandlerTest.java:103]: You should use different emails for each.

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandlerTest.java [hunk 2]
```diff
@@ -110,6 +119,34 @@ void shouldNotifyDefendant2JudgmentVariedDeterminationOfMeans_whenInvoked() {
...
             );
         }
+
+        @Test
+        void shouldNotifyRespondentLip_whenInvoked() {
+            CaseData caseData = CaseDataBuilder.builder().buildJudgmentOnlineCaseDataWithDeterminationMeans();
+            caseData = caseData.toBuilder()
+                .applicant1(Party.builder()
+                                .individualFirstName("Applicant1").individualLastName("ApplicantLastName").partyName("Applicant1")
+                                .type(Party.Type.INDIVIDUAL).partyEmail("respondentLip@example.com").build())
```
Comment (hmcts-version1-ignacio) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandlerTest.java:129]: You should use different emails for each.

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java
Comment (Omaira-Melo-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java:1]: Create a separate handler for LIP scenario

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java [hunk 3]
```diff
@@ -18,6 +18,7 @@
 
 import static java.util.Objects.nonNull;
 import static uk.gov.hmcts.reform.civil.callback.CallbackType.ABOUT_TO_SUBMIT;
+import static uk.gov.hmcts.reform.civil.callback.CaseEvent.NOTIFY_DEFENDANT1_LIP_JUDGMENT_VARIED_DETERMINATION_OF_MEANS;
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java:21]: After creation of separate callback handler we need to remove it from here?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java [hunk 4]
```diff
@@ -28,7 +29,10 @@ public class NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandle
     implements NotificationData {
 
     private static final List<CaseEvent> EVENTS = List.of(NOTIFY_SOLICITOR1_DEFENDANT_JUDGMENT_VARIED_DETERMINATION_OF_MEANS,
-                                                           NOTIFY_SOLICITOR2_DEFENDANT_JUDGMENT_VARIED_DETERMINATION_OF_MEANS);
+                                                           NOTIFY_SOLICITOR2_DEFENDANT_JUDGMENT_VARIED_DETERMINATION_OF_MEANS,
+                                                          NOTIFY_DEFENDANT1_LIP_JUDGMENT_VARIED_DETERMINATION_OF_MEANS);
```
Comment (Azam-Hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java:33]: also from here?

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java [hunk 5]
```diff
@@ -29,6 +29,8 @@ public class NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandle
 
     private static final List<CaseEvent> EVENTS = List.of(NOTIFY_SOLICITOR1_DEFENDANT_JUDGMENT_VARIED_DETERMINATION_OF_MEANS,
                                                            NOTIFY_SOLICITOR2_DEFENDANT_JUDGMENT_VARIED_DETERMINATION_OF_MEANS);
+
+    public static final String TASK_ID = "NotifyDefendantJudgmentVariedDeterminationOfMeans";
```
Comment (sjamihmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java:33]: Is TASK_ID used anywhere in this class?

**Review by bhagyashreesharma90 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java (hunk 5 above)
Comment (Omaira-Melo-Hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyDefendantJudgmentVariedDeterminationOfMeansNotificationHandler.java:33]: No, this one is not used so it was deleted

**Review by sjamihmcts — APPROVED** [review]

**Review by bhagyashreesharma90 — APPROVED** [review]

**Pull request merged**